src/Amazon.AspNetCore.Identity.Cognito/CognitoUserManager.cs

```diff
@@ -445,5 +448,55 @@ protected override Task<IdentityResult> UpdateUserAsync(TUser user)
             }
             return _userStore.UpdateAsync(user, CancellationToken);
         }
+
+        /// <summary>
+        /// Adds the specified <paramref name="claims"/> to the <paramref name="user"/>.
+        /// </summary>
+        /// <param name="user">The user to add the claim to.</param>
+        /// <param name="claims">The claims to add.</param>
+        /// <returns>
+        /// The <see cref="Task"/> that represents the asynchronous operation, containing the <see cref="IdentityResult"/>
+        /// of the operation.
+        /// </returns>
+        public override async Task<IdentityResult> AddClaimsAsync(TUser user, IEnumerable<Claim> claims)
+        {
+            ThrowIfDisposed();
+            if (user == null)
```

Comment:
Checking for user and claims to not be null is a common check. Can we make a static method to delegate to? Heck, you could even make it generic by having it take in a list of tuples (name, value).

Reply:
I would rather keep it this way for consistency with the other Identity providers:

https://github.com/aspnet/Identity/blob/eb3ff7fc32dbfff65a1ba6dfdca16487e0f6fc41/src/Microsoft.Extensions.Identity.Core/UserManager.cs#L1027

https://github.com/JudahGabriel/RavenDB.Identity/blob/e898bf6f0b33471390a8e75ce828bd4f7a305873/RavenDB.Identity/RoleStore.cs#L128

While I agree the checks may seem redundant, our codebase is already disgressing from Microsoft signatures due to the specificities of Cognito.